### PR TITLE
Set up next auth and navbar

### PR DESCRIPTION
**Pull request by benmurphyy**

### CODE REVIEW

File: frontend/components/PeerPrepNavbar.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import UserMenu from "@/components/UserMenu";
+import useUserInfo from "@/hooks/useUserInfo";
```
Comment (teoyuqi) [inline frontend/components/PeerPrepNavbar.tsx]: is it okay if we store each component in a separate directory? e.g., `PeerPrepNavbar.tsx` can be stored under
`frontend/components/PeerPrepNavbar/index.tsx` instead:
```
├── frontend
│   ├── components
│   │   ├── PeerPrepNavbar
│   │   │   ├── index.tsx      # main code here
│   │   ├── UserMenu
│   │   │   ├── index.tsx      # main code here
```

this way the code base will be neater once we have components which are more complex and have subcomponents e.g.:
```
├── frontend
│   ├── components
│   │   ├── QuestionsTable
│   │   │   ├── DifficultyTab.tsx  
│   │   │   ├── QuestionCard.tsx  
│   │   │   ├── Selectors.tsx  
│   │   │   ├── index.tsx      # main code here
...
```

File: frontend/components/Button.tsx
Comment (teoyuqi) [inline frontend/components/Button.tsx:1]: is Button.tsx included by accident haha seems like an empty file

File: frontend/components/PeerPrepNavbar.tsx (hunk 1 above)
Comment (benmurphyy) [reply frontend/components/PeerPrepNavbar.tsx]: sounds good! 👍🏻

File: frontend/components/Button.tsx
Comment (benmurphyy) [reply frontend/components/Button.tsx:1]: oh yeah HAHA, was gonna make a button but didnt need to in the end, can just remove, just merge your changes for this and prv comment straight into this branch! later we just merge this into master

**Pull request merged**